test/io/print_layout.cpp

```diff
@@ -194,3 +197,114 @@ TEST_CASE("Print wire crossing cell-level layout", "[print-cell-level-layout]")
 
     CHECK(print_stream.str() == layout_print);
 }
+
+TEST_CASE("Print empty charge layout", "[print-charge-layout]")
+{
+    using sqd_lyt = sidb_cell_clk_lyt_siqad;
+
+    constexpr const char* layout_print = "[i] empty layout\n";
+
+    std::stringstream print_stream{};
+
+    print_charge_layout(print_stream, charge_distribution_surface<sqd_lyt>{sqd_lyt{{2, 2}, "Empty"}}, false);
+
+    CHECK(print_stream.str() == layout_print);
+}
+
+TEST_CASE("Print Bestagon OR-gate", "[print-charge-layout]")
+{
+    using hex_gate_lyt = hex_even_row_gate_clk_lyt;
+
+    hex_gate_lyt layout{{1, 0}};
+
+    layout.create_or({}, {}, {0, 0});
+
+    charge_distribution_surface<sidb_cell_clk_lyt_siqad> cl{
+        convert_to_siqad_coordinates(apply_gate_library<sidb_cell_clk_lyt, sidb_bestagon_library>(layout)),
+        sidb_simulation_parameters{3, -0.32}, sidb_charge_state::NEGATIVE};
+
+    cl.assign_charge_state_by_cell_index(0, sidb_charge_state::NONE, false);
+    cl.assign_charge_state_by_cell_index(1, sidb_charge_state::NEGATIVE, false);
+    cl.assign_charge_state_by_cell_index(2, sidb_charge_state::NEUTRAL, false);
+    cl.assign_charge_state_by_cell_index(3, sidb_charge_state::NEGATIVE, false);
+    cl.assign_charge_state_by_cell_index(4, sidb_charge_state::POSITIVE, false);
+    cl.assign_charge_state_by_cell_index(5, sidb_charge_state::NEGATIVE, false);
+    cl.assign_charge_state_by_cell_index(6, sidb_charge_state::NEGATIVE, false);
+    cl.assign_charge_state_by_cell_index(7, sidb_charge_state::NEGATIVE, false);
+    cl.assign_charge_state_by_cell_index(8, sidb_charge_state::POSITIVE, false);
+    cl.assign_charge_state_by_cell_index(9, sidb_charge_state::NEGATIVE, false);
+    cl.assign_charge_state_by_cell_index(10, sidb_charge_state::NEUTRAL, false);
+    cl.assign_charge_state_by_cell_index(11, sidb_charge_state::NEGATIVE, false);
+    cl.assign_charge_state_by_cell_index(12, sidb_charge_state::NONE, false);
+    cl.assign_charge_state_by_cell_index(13, sidb_charge_state::NEGATIVE, false);
+    cl.assign_charge_state_by_cell_index(14, sidb_charge_state::NEUTRAL, false);
+    cl.assign_charge_state_by_cell_index(15, sidb_charge_state::NEGATIVE, false);
+    cl.assign_charge_state_by_cell_index(16, sidb_charge_state::POSITIVE, false);
+    cl.assign_charge_state_by_cell_index(17, sidb_charge_state::NEGATIVE, false);
```

Comment:
Same here. You could just use `assign_charge_state`

Reply:
Newest commit implements this change